scripts/convert_ima_policy.py

```diff
@@ -0,0 +1,250 @@
+    if verification_key_list:
+        policy["verification-keys"].append(verification_key_list)
+
+    return policy
+
+
+# Converts flat-format allowlist to JSON-format IMA policy
+def convert_legacy_allowlist(alist_raw):
+    ima_policy = copy.deepcopy(ima.EMPTY_IMA_POLICY)
+    ima_policy["meta"]["timestamp"] = str(datetime.datetime.now())
+    ima_policy["meta"]["generator"] = "convert-ima-policy"
+
+    lines = alist_raw.splitlines()
+    for line_num in range(lines):
+        line = lines[line_num].strip()
+        if len(line) == 0:
+            continue
+
+        pieces = line.split(None, 1)
+        if not len(pieces) == 2:
+            print(f"Line #{line_num} in Allowlist does not consist of hash and file path: {line}")
+            continue
+
+        (checksum_hash, path) = pieces
+
+        if path.startswith("%keyring:"):
+            entrytype = "keyrings"
+            path = path[len("%keyring:") :]  # remove leading '%keyring:' from path to get keyring name
+        else:
+            entrytype = "digests"
+
+        if path in ima_policy[entrytype]:
+            ima_policy[entrytype][path].append(checksum_hash)
+        else:
+            ima_policy[entrytype][path] = [checksum_hash]
+    return ima_policy
+
+
+def main():
+
+    parser = ConversionParser()
+    parser.add_argument("-a", "--allowlist", help="allowlist file location", action="store")
+    parser.add_argument("-i", "--ima_policy", help="IMA policy file location", action="store")
+    parser.add_argument("-e", "--excludelist", help="exclude list file location", action="store")
+    parser.add_argument("-v", "--verification_keys", help="list of verification key paths", action="store", type=list)
+    parser.add_argument("-o", "--output_file", help="Output file path", action="store")
+
+    # print help if no arguments provided
+    if len(sys.argv) == 1:
+        parser.print_help()
+        parser.exit()
+
+    args = parser.parse_args()
+    if (bool(args.allowlist) and not bool(args.ima_policy)) and (not bool(args.allowlist) and bool(args.ima_policy)):
+        print("Either --allowlist or --ima_policy is required!")
+        sys.exit(1)
+    elif not args.output_file:
+        print("An output file path (-o, --output_file) is required to write new policy!")
+
+    if args.allowlist:
+        policy = create_ima_policy(args.allowlist)
+    elif args.ima_policy:
+        with open(args.ima_policy, "r") as f:
+            policy = json.load(f)
+    policy_out = update_ima_policy(policy, excludelist_path=args.excludelist, verification_keys=args.verification_keys)
+    with open(args.output_file, "wb") as f:
+        f.write(json.dumps(policy_out).encode())
+
+
+class ConversionParser(argparse.ArgumentParser):
+    def error(self, message):
+        sys.stderr.write("error: %s\n" % message)
+        self.print_help(sys.stderr)
+        sys.exit(2)
+
+
+if __name__ == "__main__":
+    main()
+
+
+class TestIMAVerification(unittest.TestCase):
```

Comment:
Can we move this out?

Reply:
@aplanas (and @THS-on) Where do you propose would be the best spot to put these tests?